DEPS

```diff
@@ -69,40 +69,21 @@ hooks = [
   {
     'name': 'download_rust_deps',
     'pattern': '.',
-    'condition': 'checkout_android',
     'action': [
-      'vpython3', 'script/download_rust_deps.py', Var('brave_rust_version'), 'android'
+      'vpython3', 'script/download_rust_deps.py',
+      '--rust_version=%s' % Var('brave_rust_version'),
+      '--checkout_android=%s' % Var('checkout_android'),
+      '--checkout_ios=%s' % Var('checkout_ios'),
+      '--checkout_linux=%s' % Var('checkout_linux'),
+      '--checkout_mac=%s' % Var('checkout_mac'),
+      '--checkout_win=%s' % Var('checkout_win'),
     ]
   },
-  {
-    'name': 'download_rust_deps',
-    'pattern': '.',
-    'condition': 'checkout_mac or checkout_ios',
-    'action': [
-      'vpython3', 'script/download_rust_deps.py', Var('brave_rust_version'), 'ios'
-    ]
-  },
-  {
-    'name': 'download_rust_deps',
-    'pattern': '.',
-    'condition': 'checkout_win',
-    'action': [
-      'vpython3', 'script/download_rust_deps.py', Var('brave_rust_version'), 'win32'
-    ]
-  },
-  {
-    'name': 'download_rust_deps',
-    'pattern': '.',
-    'condition': 'checkout_linux',
-    'action': [
-      'vpython3', 'script/download_rust_deps.py', Var('brave_rust_version'), 'linux'
-    ],
-  },
   {
     # Install Web Discovery Project dependencies for Windows, Linux, and macOS
     'name': 'web_discovery_project_npm_deps',
     'pattern': '.',
-    'condition': 'not checkout_android and not checkout_ios',
+    'condition': 'checkout_linux or checkout_mac or checkout_win',
```

Comment:
this is correct condition, otherwise this is skipped if you have linux/android checkout.

Reply:
do we really need this condition at all? I use the same checkout for ios and it sometimes causes problems because I always have `target_os = ios` in my gclient file. We check out all kinds of deps that aren't used on every platform

Reply:
> do we really need this condition at all? I use the same checkout for ios and it sometimes causes problems because I always have `target_os = ios` in my gclient file. We check out all kinds of deps that aren't used on every platform

yep, we can also remove. I'm fine with that.